### PR TITLE
Ohara 1271 Implement stream app property form

### PR DESCRIPTION
**Pull request by eechih**

### CODE REVIEW

**Comment by eechih** [comment]
retry manager

File: ohara-manager/client/src/components/common/Form/Select.js [hunk 1]
```diff
@@ -68,10 +69,15 @@ const Select = ({
...
       {...rest}
     >
+      {placeholder && (
+        <option value="" disabled data-uuid="default-option">
+          {placeholder}
+        </option>
+      )}
       {isObject
-        ? list.map(({ id, name }, idx) => {
+        ? list.map(({ uuid, name }, idx) => {
```
Comment (konekoya) [inline ohara-manager/client/src/components/common/Form/Select.js]: Hmm, we should not use `uuid` here. See https://is-land.atlassian.net/browse/OHARA-1040

File: ohara-manager/client/src/components/common/FormFields/AutoSave.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+import { FormSpy } from 'react-final-form';
+import diff from 'deep-diff';
+import { debounce } from 'lodash';
+
+class AutoSave extends React.Component {
+  static propTypes = {
+    values: PropTypes.object,
+  };
+
+  constructor(props) {
```
Comment (konekoya) [inline ohara-manager/client/src/components/common/FormFields/AutoSave.js]: For consistency, let's use class properties here.

`state = { ... }`

:)

**Comment by konekoya** [comment]
run

File: ohara-manager/client/src/components/common/FormFields/AutoSave.js (hunk 2 above)
Comment (eechih) [reply ohara-manager/client/src/components/common/FormFields/AutoSave.js]: 3Q

**Comment by konekoya** [comment]
run

**Comment by konekoya** [comment]
run

**Comment by jackyoh** [comment]
retry

**Comment by konekoya** [comment]
This is neat! Thanks, @eechih

**Pull request merged**